src/Symfony/Component/Mailer/Bridge/Mailgun/RemoteEvent/MailgunPayloadConverter.php

```diff
@@ -0,0 +1,61 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Mailer\Bridge\Mailgun\RemoteEvent;
+
+use Symfony\Component\RemoteEvent\Event;
+use Symfony\Component\RemoteEvent\Event\Mailer\AbstractMailerEvent;
+use Symfony\Component\RemoteEvent\Event\Mailer\MailerDeliveryEvent;
+use Symfony\Component\RemoteEvent\Event\Mailer\MailerEngagementEvent;
+use Symfony\Component\RemoteEvent\Exception\ParseException;
+use Symfony\Component\RemoteEvent\PayloadConverterInterface;
+
+final class MailgunPayloadConverter implements PayloadConverterInterface
+{
+    /** @return AbstractMailerEvent */
+    public function convert(array $payload): Event
+    {
+        if (in_array($payload['event'], ['failed', 'accepted', 'rejected', 'delivered'])) {
+            if ('failed' === $payload['event']) {
+                $name = 'permanent' === $payload['severity'] ? MailerDeliveryEvent::BOUNCE : MailerDeliveryEvent::DEFERRED;
+            } else {
+                $name = match ($payload['event']) {
+                    'accepted' => MailerDeliveryEvent::RECEIVED,
+                    'rejected' => MailerDeliveryEvent::DROPPED,
+                    'delivered' => MailerDeliveryEvent::DELIVERED,
+                    'blocked' => MailerDeliveryEvent::DROPPED,
+                    default => throw new ParseException(sprintf('Unsupported event "%s".', $payload['event'])),
+                };
+            }
```

Comment:
Let's merge the `if` block into the `match` statement?

```suggestion
            $name = match ($payload['event']) {
                'accepted' => MailerDeliveryEvent::RECEIVED,
                'rejected' => MailerDeliveryEvent::DROPPED,
                'delivered' => MailerDeliveryEvent::DELIVERED,
                'blocked' => MailerDeliveryEvent::DROPPED,
                'failed' => 'permanent' === $payload['severity'] ? MailerDeliveryEvent::BOUNCE : MailerDeliveryEvent::DEFERRED,
                default => throw new ParseException(sprintf('Unsupported event "%s".', $payload['event'])),
            };
```

Reply:
Makes sense